KEYCLOAK-11782 Update PatternFly 3 and RCUE 3

stianst opened the pull request:
Got everything cleaned up. Split into two PRs. First cleaning up things, second updating PatternFly and RCUE.

See first commit for details on some cleanup that was made.

vmuzikar commented:
Running UI tests...

vmuzikar reviewed (changes requested):
@stianst The Admin Console somehow ... doesn't seem to work. :) Only a blank page is displayed with the following exception in the browser log:
```
angular-translate.min.js:6 Uncaught TypeError: angular.lowercase is not a function
    at N (angular-translate.min.js:6)
    at Function.V.use (angular-translate.min.js:6)
    at runTranslate (angular-translate.min.js:6)
    at Object.invoke (angular.js:5141)
    at angular.js:4940
    at r (angular.js:387)
    at fb (angular.js:4940)
    at c (angular.js:1960)
    at Object.Uc [as bootstrap] (angular.js:1981)
    at app.js:87
```

stianst committed:
KEYCLOAK-11782 Cleanup

* Updated npm-update profile
  - Update Node and NPM versions
  - Exclude optional dependencies
  - Remove node_modules within src/main/node_modules (PatternFly for example includes all dependencies in patternfly/node_modules, we don't need these)
  - Remove minimist and mkdirp from src/main/node_modules as these are not included in dist, so we don't need the source

* src/main/node_modules was updated to PatternFly 3.41.6 at some point, but not properly since src/main/resources/theme/keycloak/common/resources/node_modules/patternfly is currently on 3.38.1

* Removed ^ in versions in package.json as this results in unreliable updates of dependencies

stianst committed:
KEYCLOAK-11782 Update PatternFly 3 and RCUE 3

vmuzikar reviewed (changes requested):
@stianst Thanks for the update.

Unfortunately, this turns out to be much more complex problem than we thought. As a result neither PF nor RCUE dependency were updated in the dist.

The dependencies in dist are stored in two folders [lib](https://github.com/keycloak/keycloak/tree/b55df6d4f1d3aa4bfb5de5f4afb8d5cd7d33f13d/themes/src/main/resources/theme/keycloak/common/resources/lib) and [node_modules](https://github.com/keycloak/keycloak/tree/b55df6d4f1d3aa4bfb5de5f4afb8d5cd7d33f13d/themes/src/main/resources/theme/keycloak/common/resources/node_modules) and e.g. PF is placed in both but only `node_modules` is updated by this PR. However, PF [is loaded from `lib`](https://github.com/keycloak/keycloak/blob/b55df6d4f1d3aa4bfb5de5f4afb8d5cd7d33f13d/themes/src/main/resources/theme/keycloak/admin/theme.properties#L3) and therefore e.g. Admin Console still uses the old PF version. Things are even more complicated because RH-SSO includes its own [folder with dependencies](https://github.com/keycloak/keycloak/tree/b55df6d4f1d3aa4bfb5de5f4afb8d5cd7d33f13d/themes/src/main/resources-product/theme/rh-sso/common/resources/lib) where RCUE is stored (it's [loaded from this folder as well](https://github.com/keycloak/keycloak/blob/b55df6d4f1d3aa4bfb5de5f4afb8d5cd7d33f13d/themes/src/main/resources-product/theme/rh-sso/admin/theme.properties#L3)).

I've done some quick testing as there's no point in testing it more deeply and I've found a weird bug with login page background – see the screenshot. Maybe it's somehow caused by the inconsistencies I mentioned above.
![Screenshot 2019-10-29 at 08 40 53](https://user-images.githubusercontent.com/13906855/67770666-1e5cae80-fa57-11e9-8987-7d9bf0aa7883.png)

abstractj commented:
@stianst @vmuzikar the issues with the login screen where gone for me. All the tests for the `console` are passing and I just found some errors 
 while running `base-ui` tests, but it seems unrelated with your changes:

```
[ERROR] Failures:
[ERROR]   RequiredActionsTest.clientConsent:318 Expected and actual Client Scopes to approve don't match
[ERROR] Errors:
[ERROR]   LoginPageTest.resetCredentialsTest:284 » ElementNotInteractable element not in...
[ERROR]   RequiredActionsTest.configureManualHotp:290->testManualOtp:342 » ElementNotInteractable
[ERROR]   RequiredActionsTest.configureManualTotp:284->testManualOtp:355 » NoSuchElement
```

Depending on what @vmuzikar says, we can merge it.

stianst committed:
KEYCLOAK-11844 Update account theme to use shared PatternFly/RCUE

stianst committed:
KEYCLOAK-11845 Update admin theme to use shared PatternFly/RCUE

stianst commented:
Figured out the problem with account and admin. They where missing patternfly/rcue-additions so update everything now and removed PF/RCUE from lib

vmuzikar commented:
@stianst Ok, glad you figured it out. :) Is this ready for re-review?

stianst commented:
@vmuzikar yes, should be ready to review now

vmuzikar reviewed (changes requested):
@stianst RH-SSO Admin Console theme is completely broken. :(
![Screenshot 2019-10-31 at 14 36 04](https://user-images.githubusercontent.com/13906855/67951473-d91db580-fbeb-11e9-92b4-282b7645efe6.png)

vmuzikar reviewed (changes requested):
@stianst Thanks for the update!

I've found one last regression (sorry for that :D). The error notifications both in Admin and Account Console are now missing the red cross icon. It really affects just the errors – other notifications have their icons.

stianst commented:
@vmuzikar Fixed. Looks like PatternFly changed the icons slightly. The last change only affects warning and error notifications.

vmuzikar reviewed (changes requested):
@stianst The Account Console is still affected – no icon in error notifications.

stianst commented:
@vmuzikar somehow I missed that you wrote both admin console and account console. Fixed the account console now. Crossing fingers we're done now

vmuzikar reviewed (approved):
@stianst Yes, we're done, LGTM now! Thank you for your patience with my requests. :)
